### ISSUE TITLE
Occasional NPE after stopPlayback has been called

### ISSUE THREAD

**Issue by inmyth** [niqdev/ipcam-view#22]
So I put stopPlayback in fragment's onPause which activates during fragment change. Occasionally an NPE is thrown:
```
java.lang.NullPointerException: Attempt to invoke virtual method  
android.graphics.Bitmap`com.github.niqdev.mjpeg.MjpegInputStreamDefault.readMjpegFrame()
on a null object reference at 
com.github.niqdev.mjpeg.MjpegViewDefault$MjpegViewThread.run(MjpegViewDefault.java:121)
```
The said part of the code is :
``` 
   while (mRun) {
           if (surfaceDone) {
                  try {
                       c = mSurfaceHolder.lockCanvas();
                        synchronized (mSurfaceHolder) {
                            try {
                               bm = mIn.readMjpegFrame(); //where NPE happens
...
```
I feel that even after mRun is set to false with ```stopPlayback```, the process inside has not finished while mIn is already null. What would you say of adding a null check for mIn ?

**Comment by niqdev** [comment]
Hi, unfortunately there are so many issue related to the wrapped implementation and I have no time now, but I'd really like to rewrite the library from scratch, as pointed out in the issue #18 it seems that version `0.3.4` is more stable.
Let me know if is the same for you and I should revert some changes, if you have any suggestion to fix the problem or you are brave enough to dig into the implementation I'll be more than happy to merge any pull request!

**Comment by John-Keyor** [comment]
Hi,
Yeah, I've been having that same problem for few months now. Even tried changing the dependency in build.gradle to 0.3.4 from 0.3.6 and same problem. It isn't always reproducible but more often than not, when exiting from videoplayback, there is a NPE. 

I'd really like to continue implementing ipcam-mjpeg view but with this problem, it's become a nuisance. Is there anyway on the implementation side to fix this?

**Comment by niqdev** [comment]
Hi @John-Keyor,
as I said I spent already too much time trying to debug/fix it without success. The problem seems to be the bad Thread management / convoluted / messy codes that this library wraps. Unfortunately I have no spare time to invest on this project in the near future to rewrite it, but I would be more than happy if someone wanna help me.
You could try something different using the `native` implementation, I'm not an expert of C and before going on to add extra complexity I prefered to pause it.
I'm currently using this library in this [project](https://github.com/openwebnet/openwebnet-android) [here](https://github.com/openwebnet/openwebnet-android/blob/master/app/src/main/java/com/github/openwebnet/view/device/IpcamStreamActivity.java)

Everybody is more than welcome to contribute to the project or to propose any idea to improve it.

**Comment by niqdev** [comment]
Hi @John-Keyor  and @inmyth , I just published version `0.4.0` to solve and issue to manage multiple view in the same activity. Could you please tell me if with the new version you still have the same probelms?
Thanks

**Comment by niqdev** [comment]
Since I didn't receive any updates I'll close the issue
